kclvm/tools/src/query/override.rs

```diff
@@ -1,50 +1,242 @@
+///     age = 10
+/// }
+/// ```
+///
+/// - test.k (after override)
+///
+/// ```kcl
+/// schema Person:
+///     age: int
+///
+/// alice = Person {
+///     age = 18
+/// }
+/// ```
+pub fn override_file(file: &str, specs: &[String], import_paths: &[String]) -> Result<bool> {
+    let overrides = specs
+        .iter()
+        .map(|s| parse_override_spec(s))
+        .filter_map(Result::ok)
+        .collect::<Vec<ast::OverrideSpec>>();
+    let mut module = match parse_file(file, None) {
+        Ok(module) => module,
+        Err(msg) => return Err(anyhow!("{}", msg)),
+    };
+    let mut result = false;
+    for o in &overrides {
+        if apply_override_on_module(&mut module, o, import_paths)? {
+            result = true;
+        }
+    }
+    if result {
+        let code_str = print_ast_module(&module);
+        std::fs::write(file, &code_str)?
+    }
+    Ok(result)
+}
+
+/// Override spec string to override structure
+///
+/// # Example
+///
+/// ```
+/// use kclvm_tools::query::parse_override_spec;
+/// use kclvm_ast::ast;
+///
+/// let spec = parse_override_spec("alice.age=10").unwrap();
+/// assert_eq!(spec, ast::OverrideSpec {
+///     pkgpath: "".to_string(),
+///     field_path: "alice.age".to_string(),
+///     field_value: "10".to_string(),
+///     action: ast::OverrideAction::CreateOrUpdate,
+/// });
+/// ```
+pub fn parse_override_spec(spec: &str) -> Result<ast::OverrideSpec> {
+    let err = Err(invalid_spec_error(spec));
+    if spec.contains('=') {
+        // Create or update the override value.
+        let split_values = spec.splitn(2, '=').collect::<Vec<&str>>();
+        let path = split_values
+            .get(0)
+            .ok_or_else(|| invalid_spec_error(spec))?;
+        let field_value = split_values
+            .get(1)
+            .ok_or_else(|| invalid_spec_error(spec))?;
+        let (pkgpath, field_path) = split_field_path(path)?;
+        Ok(ast::OverrideSpec {
+            pkgpath,
+            field_path,
+            field_value: field_value.to_string(),
+            action: ast::OverrideAction::CreateOrUpdate,
+        })
+    } else if let Some(stripped_spec) = spec.strip_suffix('-') {
+        // Delete the override value.
+        let (pkgpath, field_path) = split_field_path(stripped_spec)?;
+        Ok(ast::OverrideSpec {
+            pkgpath,
+            field_path,
+            field_value: "".to_string(),
+            action: ast::OverrideAction::Delete,
+        })
+    } else {
+        err
+    }
+}
 
-pub struct OverrideInfo {
-    pub pkgpath: String,
-    pub filename: String,
-    pub module: ast::Module,
+/// Get field package path and identifier name from the path.
+///
+/// # Example
+///
+/// ```
+/// use kclvm_tools::query::split_field_path;
+///
+/// let (pkgpath, field) = split_field_path("pkg.to.path:field").unwrap();
+/// assert_eq!(pkgpath, "pkg.to.path");
+/// assert_eq!(field, "field");
+/// ```
+pub fn split_field_path(path: &str) -> Result<(String, String)> {
```

Comment:
This method looks like a generic util?

Reply:
> This method looks like a generic util?

Move it to the `util` mod.

Reply:
I don't think the  helpers' implementation of the language specification should go in the query package. I think it should be a package of the lang-spec mechanism, which specifically implements helpers/utils/tools heavily associated with various language designs and spec, and we should maintains it uniformly to avoid fragmentation. 

Cause this matter is beyond the scope of this PR, you can leave a todo and issue for now, and handle it later.